### ISSUE TITLE
Define a better ActionController interface.

### ISSUE THREAD

**Issue by sirwolfgang** [sirwolfgang/jwt_keeper#26]
I am not really happy with the action controller interface; but I am not entirely sure what a better interface would look like.

**Comment by milankubin** [comment]
It would be nice to have better integration with other authentication flows, and multi-model configuration that's a little bit easier to set.
I have looked at several JWT modules, and i specifically thought out I needed to use Redis + jti to have a kind of stateful JWT to limit sessions 1 per device, and be able to revoke the token/session two ways, but also not hitting the DB for every auth. Redis is a little faster for that. So happy I found your module that had a nice JTI + Redis implementation.

My use case  is: user signs in from mobile app through normal u/p (devise), gets 10 min token assigned, then is allowed to list "teams" scheduled for that day, then log in on the selected team where they get a 24hour token.  (This is "ok" since we're also whitelisting device UUID's and session limiting to 1).
The problem is however that with the current interface I would need to JWTKeeper.configure to override the defaults (like expiry) in the controller, and then overload the methods  authenticated/not_authenticated in every controller, because :require_authentication does not allow to pass through the object I would like to work with.

It would be nice to have something like :require_authentication_for_model   or even just by passing a param like:
 before_action { |a| require_authentication Model }
And then look inside the model for config overrides
"jwt_keeper_authenticatable  expiry: 10.minutes, issuer: ... "

This would allow me to keep it DRI and just overload one time inside my APIController and do stuff based on the object i'm working with.

**Comment by sirwolfgang** [comment]
Can please link to some of the controller interfaces you liked?

[Side Note: The cookie locking feature would also functional secure a session to a device]

You do not need to override `JWTKeeper.configure`, anything you pass in when creating a token will take precedence via `JWTKeeper::Token.create(token_claims)`. With this in mind I not sure I understand the need for the additional interface.

**Comment by milankubin** [comment]
I don't really see how? Maybe i'm missing something really obvious.
But it looks like configuration is set and frozen on initialize.
Any  config hash passed will merge with defaults.
Thus, when I want to just set the expiry for a specific token, It seems I need to config all the values + the desired expiry, otherwise configure will just merge with set defaults and freeze the hash.

Also Token.create does not accept config params ?

````ruby
# File 'lib/jwt_keeper/token.rb', line 21

def self.create(private_claims)
  cookie_secret = SecureRandom.hex(16) if JWTKeeper.configuration.cookie_lock
  new(private_claims, cookie_secret)
end
`````

**Comment by sirwolfgang** [comment]
Expiration like everything else is a claim. So you just create the token with the overriding claim. 
`Token.create(exp: 5.minutes.from_now.to_i)`

The since the private claims are applied last, they will overwrite any other claims set by default or config. If you don't override the claim, then the default value will still be there. I added a spec to illustrate, and protect, this feature: https://github.com/sirwolfgang/jwt_keeper/blob/develop/spec/lib/jwt_keeper/token_spec.rb#L21-L26

This also works with rotation. So when you want to switch to 24 hour token, you can rotate into it; Reducing the code needed for worrying about revoking the old token.

**Comment by milankubin** [comment]
Aha, I didn't realise that thanks! I went through all the code, and noticed it would work for rotation, but it wasn't clear to me that it would work through claims to.
Maybe my requirements are a little bit more out of scope of this gem, but I would prefer to depend on redis-store / redis-rails so it will go through the native Session handler interface instead of using a direct redis_connection.  This way i'll also be able to create a gem that will work as a device module.
I'll fork off for a redis-store depending integration anyway, and maybe you want to pull it.  Let me know ! Thanks fir this great gem. I got the integration working fine in my use case!
